addons/website/static/src/snippets/s_searchbar/options.js

```diff
@@ -22,6 +22,18 @@ options.registry.SearchBar = options.Class.extend({
                     defaultOrder.click(); // close
                 }
             }});
+
+            // Reset display options.
+            const displayOptions = new Set();
```

Comment:
@bso-odoo 
I saw that you marked the other review as processed + push forced the branch 3 days ago but didn't reply to this one.
Are you still processing this one or forgot about it or did we discuss about it and I am the one who forgot about it :) ?
just to be sure..

Reply:
@rdeodoo I am sure I wrote an answer to this... I must have forgotten to confirm it. :man_facepalming: Sorry...

The reason for resetting is because when returning to modes that have more options, it feels even more wrong if the option appears disabled.
When transitioning from "Everything" to "Blog", they must be disabled otherwise there is something with a wrong state that the UI does not let you modify.
When transitioning from "Blog" to "Everything", the "Image" and "Extra Link" options would reappear but become disabled.

If you want, I can try looking into keeping the option's value if it was visible and remains visible.

Reply:
I was speaking about the whole js block, I don't understand why we need that, see video: https://watch.screencastify.com/v/YfwAklyhfAnjXaKG3ctH

There is nothing wrong **functionally** here, right? So that's why I am confused about why we need this, what is the technical bug fixed here?

Reply:
There is no problem during the edition itself, but it is strange if you use the search in the various states.
You end up with two kinds of search results for a "blog only" search, either with images or without images - even though while configuring the blog search the "Image" option is not available.

Without the reset:
![image](https://user-images.githubusercontent.com/71644421/200011163-061806b2-612c-476f-96db-9e44f80b3d6b.png)

With the reset (or by turning it off before switching to blog):
![image](https://user-images.githubusercontent.com/71644421/200011284-33a37fd3-de68-4ca9-bb3c-97b37c8ade64.png)